src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md

```diff
@@ -0,0 +1,712 @@
+}
+```
+
+</div>
+
+
+## Arrival rate
+
+Up until v0.27.0 k6 only supported a closed model for the simulation of new VU arrivals,
+or expressed in another way, the start of new VU iterations.
+In a closed model a new VU iteration only starts when a VU's previous iteration has
+completed executing. The start rate, or arrival rate, of new VU iterations in a closed
+model is thus tightly coupled with the iteration duration (time from start to finish
+of the VU's `exec` function, by default the `export default function`):
+
+<div class="code-group" data-props='{"labels": [ "closed-model.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  scenarios: {
+    closed_model: {
+      executor: 'constant-vus',
+      vus: 1,
+      duration: '1m',
+    }
+  }
+};
+
+export default function() {
+  // The following request will take roughly 6s to complete,
+  // resulting in an iteration duration of 6s.
+  // New VU iterations will thus start at a rate of 1 per 6s,
+  // and we can expect to get 10 iterations completed in total for the
+  // full 1m test duration.
+  http.get('http://httpbin.test.k6.io/delay/6');
+}
+```
+
+</div>
+
+Running this script would result in something like:
+
+```
+running (1m01.5s), 0/1 VUs, 10 complete and 0 interrupted iterations
+closed_model ✓ [======================================] 1 VUs  1m0s
+```
+
+This tight coupling between the VU iteration duration and start of new VU iterations
+in effect means that the target system can influence the throughput of the test, via
+its response time. Slower response times means longer iterations and a lower arrival
+rate of new iterations, and vice versa for faster response times.
+
+In other words, when the target system is being stressed and starts to respond more
+slowly a closed model load test will play "nice" and wait, resulting in increased
+iteration durations and a tapering off of the arrival rate of new VU iterations.
+
+This is not ideal when the goal is to simulate a certain arrival rate of new VUs,
+or more generally throughput (eg. requests per second).
+
+To fix this problem we use an open model, decoupling the start of new VU iterations
+from the iteration duration and the influence of the target system's response time.
+
+![Arrival rate closed/open models](images/arrival-rate-open-closed-model.png)
+
+In k6 we've implemented this open model with our "arrival rate" executors. There are
+two arrival rate executors to chose from for your scenario(s),
+[constant-arrival-rate](#constant-arrival-rate) and [ramping-arrival-rate](#ramping-arrival-rate):
+
+<div class="code-group" data-props='{"labels": [ "open-model.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  scenarios: {
+    open_model: {
+      executor: 'constant-arrival-rate',
+      rate: 1,
+      timeUnit: '1s',
+      duration: '1m',
+      preAllocatedVUs: 1,
+      maxVUs: 100,
```

Comment:
Allocating VUs in the middle of the test is probably a bad practice in general. While we support it, we shouldn't give example where it's obvious that the preAllocatedVUs will be very insufficient.
```suggestion
      preAllocatedVUs: 20,
      maxVUs: 100,
```